go/libraries/doltcore/sqle/dolt_diff_table_function.go

```diff
@@ -127,29 +142,33 @@ func (dtf *DiffTableFunction) RowIter(ctx *sql.Context, _ sql.Row) (sql.RowIter,
 	// TODO: When we add support for joining on table functions, we'll need to evaluate this against the
 	//       specified row. That row is what has the left_table context in a join query.
 	//       This will expand the test cases we need to cover significantly.
-	fromCommit, toCommit, _, err := dtf.evaluateArguments()
+	fromCommitVal, toCommitVal, dotCommitVal, _, err := dtf.evaluateArguments()
 	if err != nil {
 		return nil, err
 	}
 
-	fromHash, fromOk := fromCommit.(string)
-	toHash, toOk := toCommit.(string)
-	if !fromOk || !toOk {
-		return nil, fmt.Errorf("expected strings for from and to revisions, got: %v, %v", fromHash, toHash)
-	}
-
 	sqledb, ok := dtf.database.(Database)
 	if !ok {
-		panic("unable to get dolt database")
+		return nil, fmt.Errorf("unable to get dolt database")
 	}
-	ddb := sqledb.GetDoltDB()
 
-	dp := dtables.NewDiffPartition(dtf.tableDelta.ToTable, dtf.tableDelta.FromTable, toHash, fromHash, dtf.toDate, dtf.fromDate, dtf.tableDelta.ToSch, dtf.tableDelta.FromSch)
+	fromCommitStr, toCommitStr, err := loadCommitStrings(ctx, fromCommitVal, toCommitVal, dotCommitVal, sqledb)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(fromCommitStr) == 0 || len(toCommitStr) == 0 {
+		return nil, fmt.Errorf("expected strings for from and to revisions, got: %v, %v", fromCommitStr, toCommitStr)
+	}
```

Comment:
Should `loadCommitStrings` throw this error? It looks like we do this check both times `loadCommitStrings` is called, although I do see you are using different error messages. Just wondering if it's possible to fold them into the same error message and not have to check this in two places.